risc0/zkvm/src/guest/env.rs

```diff
@@ -322,10 +323,15 @@ pub fn verify_assumption(claim: Digest, control_root: Digest) -> Result<(), Infa
 /// receives the return data.
 ///
 /// On the host side, implement SliceIo to provide a handler for this call.
+///
+/// NOTE: This method never frees up the buffer memory storing the host's response.
 pub fn send_recv_slice<T: Pod, U: Pod>(syscall_name: SyscallName, to_host: &[T]) -> &'static [U] {
     let syscall::Return(nbytes, _) = syscall(syscall_name, bytemuck::cast_slice(to_host), &mut []);
     let nwords = align_up(nbytes as usize, WORD_SIZE) / WORD_SIZE;
-    let from_host_buf = unsafe { core::slice::from_raw_parts_mut(sys_alloc_words(nwords), nwords) };
+    let from_host_buf = unsafe {
+        let layout = Layout::from_size_align(nwords * WORD_SIZE, 4).unwrap();
```

Comment:
Perhaps creating a `WORD_ALIGN` constant so that when we support 64-bit this gets automatically updated?

Reply:
possibly, but might be out of scope for this PR since this alignment of 4 is also explicitly used in the c-kzg example and the sha2 accelerator?